Fix deprecation warnings from nightly Drake

After building with Drake's nightly binaries on Focal 20.04 on `rolling`, I encountered a few Drake API deprecation warnings in  RosSubscriberSystem:


```
--- stderr: drake_ros_core
/home/robot/workspace/src/drake_ros_core/ros_subscriber_system.cpp: In constructor ‘drake_ros_core::RosSubscriberSystem::RosSubscriberSystem(std::unique_ptr<drake_ros_core::SerializerInterface>&, const string&, const rclcpp::QoS&, std::shared_ptr<drake_ros_core::DrakeRosInterface>)’:
/home/robot/workspace/src/drake_ros_core/src/ros_subscriber_system.cpp:74:6: error: ‘drake::systems::LeafOutputPort<T>& drake::systems::LeafSystem<T>::DeclareAbstractOutputPort(typename drake::systems::LeafOutputPort<T>::AllocCallback, typename drake::systems::LeafOutputPort<T>::CalcCallback, std::set<drake::TypeSafeIndex<drake::systems::DependencyTag> >) [with T = double; typename drake::systems::LeafOutputPort<T>::AllocCallback = std::function<std::unique_ptr<drake::AbstractValue>()>; typename drake::systems::LeafOutputPort<T>::CalcCallback = std::function<void(const drake::systems::Context<double>&, drake::AbstractValue*)>]’ is deprecated: \nDRAKE DEPRECATED: Pass a port name as the first argument.\nThe deprecated code will be removed from Drake on or after 2021-10-01. [-Werror=deprecated-declarations]
   74 |     });
      |      ^
In file included from /home/robot/workspace/src/drake_ros_core/include/drake_ros_core/ros_subscriber_system.hpp:17,
                 from /home/robot/workspace/src/drake_ros_core/src/ros_subscriber_system.cpp:21:
/opt/drake/include/drake/systems/framework/leaf_system.h:1655:22: note: declared here
 1655 |   LeafOutputPort<T>& DeclareAbstractOutputPort(
      |                      ^~~~~~~~~~~~~~~~~~~~~~~~~
/home/robot/workspace/src/drake_ros_core/src/ros_subscriber_system.cpp: In member function ‘virtual void drake_ros_core::RosSubscriberSystem::DoCalcNextUpdateTime(const drake::systems::Context<double>&, drake::systems::CompositeEventCollection<double>*, double*) const’:
/home/robot/workspace/src/drake_ros_core/src/ros_subscriber_system.cpp:128:53: error: ‘void drake::systems::EventCollection<EventType>::add_event(std::unique_ptr<_Tp>) [with EventType = drake::systems::UnrestrictedUpdateEvent<double>]’ is deprecated: \nDRAKE DEPRECATED: Use AddEvent instead.\nThe deprecated code will be removed from Drake on or after 2021-09-01. [-Werror=deprecated-declarations]
  128 |       drake::systems::TriggerType::kTimed, callback));
      |                                                     ^
In file included from /opt/drake/include/drake/systems/framework/system.h:27,
                 from /opt/drake/include/drake/systems/framework/leaf_system.h:29,
                 from /home/robot/workspace/src/drake_ros_core/include/drake_ros_core/ros_subscriber_system.hpp:17,
                 from /home/robot/workspace/src/drake_ros_core/src/ros_subscriber_system.cpp:21:
/opt/drake/include/drake/systems/framework/event_collection.h:138:16: note: declared here
  138 |   virtual void add_event(std::unique_ptr<EventType> event) = 0;
      |                ^~~~~~~~~
cc1plus: all warnings being treated as errors

```

I will be updating this repo with a Dockerized build and CI to catch these warnings.

--- a/drake_ros_core/src/ros_subscriber_system.cpp
+++ b/drake_ros_core/src/ros_subscriber_system.cpp
@@ -67,7 +67,6 @@ RosSubscriberSystem::RosSubscriberSystem(
     std::bind(&RosSubscriberSystemPrivate::handle_message, impl_.get(), std::placeholders::_1));
 
   DeclareAbstractOutputPort(
-    topic_name,
     [serializer{impl_->serializer_.get()}]() {return serializer->create_default_value();},
     [](const drake::systems::Context<double> & context, drake::AbstractValue * output_value) {
       // Transfer message from state to output port

--- a/drake_ros_core/src/ros_subscriber_system.cpp
+++ b/drake_ros_core/src/ros_subscriber_system.cpp
@@ -67,6 +67,7 @@ RosSubscriberSystem::RosSubscriberSystem(
     std::bind(&RosSubscriberSystemPrivate::handle_message, impl_.get(), std::placeholders::_1));
 
   DeclareAbstractOutputPort(
+    topic_name,
     [serializer{impl_->serializer_.get()}]() {return serializer->create_default_value();},
     [](const drake::systems::Context<double> & context, drake::AbstractValue * output_value) {
       // Transfer message from state to output port
@@ -123,8 +124,8 @@ RosSubscriberSystem::DoCalcNextUpdateTime(
   *time = context.get_time();
   drake::systems::EventCollection<drake::systems::UnrestrictedUpdateEvent<double>> & uu_events =
     events->get_mutable_unrestricted_update_events();
-  uu_events.add_event(
-    std::make_unique<drake::systems::UnrestrictedUpdateEvent<double>>(
+  uu_events.AddEvent(
+    drake::systems::UnrestrictedUpdateEvent<double>(
       drake::systems::TriggerType::kTimed, callback));
 }
 }  // namespace drake_ros_core

